fix: azure blob file not found

IsQiao opened the pull request:
Fixes #11673

Skrypt commented:
The fix is questionnable. Needs investigation.

deanmarcussen commented on src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:
```diff
@@ -145,7 +145,7 @@ await foreach (var blob in page)
                 else
                 {
-                    var itemName = Path.GetFileName(WebUtility.UrlDecode(blob.Blob.Name)).Trim('/');
+                    var itemName = Path.GetFileName(blob.Blob.Name).Trim('/');
```
The decode code is old from the previous blob library from Microsoft.

It maybe that it is no longer delivered by the azure library encoded, so a decode doesn't do anything useful.

IsQiao replied:
the `decode` can damage file path. `aaa+++.jpg` -> `aaa   .jpg`
remove `decode` can resolve it if use Kestrel Server directly.
but need to set `allowDoubleEscaping="true"` in web.config if use IIS Server.

Piedone commented on src/OrchardCore/OrchardCore.FileStorage.AzureBlob/BlobFileStore.cs:
Please check the decode in `GetDirectoryContentFlatAsync()` too.

While this indeed fixes the linked bug, I'm not sure if it'll break anything else. Can you upload a file to Blob Storage with a name that has all kinds of characters that should be encoded for a URL, and it'll keep them in their original form?

IsQiao replied:
ok, i will to test all kinds of chars for this case and resolve the PR conflicts. please wait.

Piedone replied:
Thank you!

Piedone reviewed (changes requested):
I sincerely apologize for us taking so much time here. I checked out the PR, going over old ones, and this would be quite useful.

Is this something you'd like to revisit any time soon or should we close?

Piedone commented:
Not to be a bother, but any news here, @IsQiao?

IsQiao commented:
Sorry, I found new bugs regarding this fix and I will fix them.

Piedone commented:
Great, thanks! When you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

IsQiao commented:
> Great, thanks! When you're done addressing all feedback of a review, click ["Re-request review"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#re-requesting-a-review) in the top-right corner for each reviewer when you're ready for another round of review, so they know that you're done.

ok, i will click the button after fix the bug. thanks

Piedone commented:
This needs to be fixed not just for Blob Storage, see https://github.com/OrchardCMS/OrchardCore/issues/11673#issuecomment-2134038340.
